Mathlib/GroupTheory/Congruence/Basic.lean

```diff
@@ -69,6 +69,61 @@ protected def congr {c d : Con M} (h : c = d) : c.Quotient ≃* d.Quotient :=
   { Quotient.congr (Equiv.refl M) <| by apply Con.ext_iff.mp h with
     map_mul' := fun x y => by rcases x with ⟨⟩; rcases y with ⟨⟩; rfl }
 
+@[simp]
+theorem congr_mk {c d : Con M} (h : c = d) (a : M) :
+    Con.congr h (a : c.Quotient) = (a : d.Quotient) := rfl
+
+@[to_additive]
+theorem comap_conGen_of_Bijective {M N : Type*} [Mul M] [Mul N] (f : M → N)
+    (hf : Function.Bijective f) (H : ∀ (x y : M), f (x * y) = f x * f y) (rel : N → N → Prop) :
+    Con.comap f H (conGen rel) = conGen (fun x y ↦ rel (f x) (f y)) := by
```

Comment:
Can you prove an inequality one way using only injectivity/surjectivity?

Reply:
Neat idea! I can prove one inequality without either, and then the other inequality requires bijectivity. I've separated out the inequality which doesn't require either as conGen_le_comap; but when I use it for the second half of comap_conGen_of_Bijective, it only unifies when I use "apply", not "exact" --
"show term" produces an "exact" statement that won't work. Do you have any idea what's happening?

Reply:
Thanks! I think I might've figured it out.

I think it's because of the implicit arguments for the instance of LE on Con
∀ ⦃x y⦄, c x y → d x y

(which here should be a and b)

Lean won't match them correctly with "exact"; it searches harder with "apply"

If I simply give them to "exact", by using @ to specify the implicit arguments as well, that works too

And even better, if I use a fact about LE to avoid using ext, everything clicks into place just right

Thanks for pointing out that the two directions should be separate - it makes the code much nicer!